src/Disks/S3/DiskS3.cpp

```diff
@@ -1002,6 +1002,14 @@ void DiskS3::restoreFileOperations(const RestoreInformation & restore_informatio
             else
                 to_path /= from_path.filename();
 
+            /// to_path may exist and non-empty in case for example abrupt restart, so remove it before rename
+            if (metadata_disk->exists(to_path))
+            {
+                if (metadata_disk->isFile(to_path))
```

Comment:
I think this check is redundant since we know that this is a directory.

Reply:
OK, If no one else create a file named `to_path`, the only case is abrupt restart and it must be a directory.